specs/governance/gov-token.md

```diff
@@ -24,72 +24,75 @@
 
 `GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
 `ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
-voting system.
+voting system. The contract integrates with the `Alligator` contract through a hook-based approach to support subdelegations.
 
-The specific implementation details of the voting and delegation logic are inherited from the `ERC20Votes` contract. The
-`GovernanceToken` contract focuses on integrating this functionality with `ERC20Burnable` and adding the minting
-capability. EIP-712 permit functionality is also inherited from `ERC20Permit`, which is imported by `ERC20Votes`.
+### Hook-based Integration with Alligator
 
-### Checkpoints
+Subdelegations allow for advanced delegation use cases, such as partial, time-constrained & block-based delegations, relative
+& fixed allowances, and custom rules. The `_afterTokenTransfer` function in the `GovernanceToken` is modified to call the
+`afterTokenTransfer` function in the `Alligator` contract, allowing the `Alligator` contract to consume the hooks and update
+its delegation and checkpoint mappings accordingly.
 
-`ERC20Votes` uses checkpoints to track voting power history. A checkpoint is a "snapshot" of the voting power of an
-address at a specific block number.
+If the call to the `Alligator`'s `afterTokenTransfer` function fails, the token transfer must still be completed, and the
+following `AlligatorCallFailed` event will be emitted to indicate the failed call:
 
 ```solidity
-struct Checkpoint {
-    uint32 fromBlock;
-    uint224 votes;
-}
+event AlligatorCallFailed(address indexed transferFrom, address indexed transferTo, uint256 amount);
 ```
 
-Checkpoints are organized as a one-to-one mapping of user addresses to append-only lists of checkpoints. When a
-user is involved in a token transfer, mint, burn, or vote delegation, a new checkpoint is added to their corresponding
-list of checkpoints, reflecting the user's updated voting power at that block number.
+This prevents the token transfer from being blocked due to issues with the `Alligator` contract.
 
-```solidity
-mapping(address => Checkpoint[]) private _checkpoints;
-```
+All delegation-related state, including the `delegates`, `checkpoints`, and `numCheckpoints` mappings, is gradually
+shifted from the `GovernanceToken` to the `Alligator` contract through transactions that call the `Alligator`'s hook
+(i.e., transfers). In the hook, the `Alligator` should check if the `to` and `from` addresses have been migrated.
```

Comment:
major nit cause this is a random thing I know about 😅

here it should be "e.g." instead of "i.e.", since the former means "for example" and is non-comprehensive and the latter means "that is" and denotes equivalence
```suggestion
(e.g. transfers). In the hook, the `Alligator` should check if the `to` and `from` addresses have been migrated.
```

Reply:
yup, good catch!